src/test/e2e-emulated/map.spec.ts

```diff
@@ -1307,4 +1307,2282 @@ describe("map", () => {
+                expect((await contract.getIntMap5()).size).toBe(1);
+                expect((await contract.getIntMap6_1()).size).toBe(1);
+                expect((await contract.getIntMap6_2()).size).toBe(1);
+                expect((await contract.getIntMap6_3()).size).toBe(1);
+                expect((await contract.getIntMap6_4()).size).toBe(1);
+                expect((await contract.getIntMap6_5()).size).toBe(1);
+                expect((await contract.getIntMap6_6()).size).toBe(1);
+                expect((await contract.getIntMap6_7()).size).toBe(1);
+                expect((await contract.getIntMap7_1()).size).toBe(1);
+                expect((await contract.getIntMap7_2()).size).toBe(1);
+                expect((await contract.getIntMap7_3()).size).toBe(1);
+                expect((await contract.getIntMap7_4()).size).toBe(1);
+                expect((await contract.getIntMap7_5()).size).toBe(1);
+                expect((await contract.getIntMap7_6()).size).toBe(1);
+                expect((await contract.getIntMap8_1()).size).toBe(1);
+                expect((await contract.getIntMap8_2()).size).toBe(1);
+                expect((await contract.getIntMap8_3()).size).toBe(1);
+                expect((await contract.getIntMap8_4()).size).toBe(1);
+                expect((await contract.getIntMap8_5()).size).toBe(1);
+                expect((await contract.getIntMap8_6()).size).toBe(1);
+                expect((await contract.getIntMap8_7()).size).toBe(1);
+                expect((await contract.getIntMap9_1()).size).toBe(1);
+                expect((await contract.getIntMap9_2()).size).toBe(1);
+                expect((await contract.getIntMap9_3()).size).toBe(1);
+                expect((await contract.getIntMap9_4()).size).toBe(1);
+                expect((await contract.getIntMap9_5()).size).toBe(1);
+                expect((await contract.getIntMap9_6()).size).toBe(1);
+                expect((await contract.getIntMap10_1()).size).toBe(1);
+                expect((await contract.getIntMap10_2()).size).toBe(1);
+                expect((await contract.getIntMap10_3()).size).toBe(1);
+                expect((await contract.getIntMap10_4()).size).toBe(1);
+                expect((await contract.getAddrMap1()).size).toBe(1);
+                expect((await contract.getAddrMap2()).size).toBe(1);
+                expect((await contract.getAddrMap3()).size).toBe(1);
+                expect((await contract.getAddrMap4()).size).toBe(1);
+
+                // Check .replace return value
+                expect(await contract.getIntMap1Replace(k, null)).toBe(true);
+                expect(await contract.getIntMap1Replace(k + 1n, null)).toBe(
+                    false,
+                );
+                expect(await contract.getIntMap1Replace(k, 123n)).toBe(true);
+                expect(await contract.getIntMap1Replace(k + 1n, 123n)).toBe(
+                    false,
+                );
```

Comment:
I think that further tests need to be added here for testing the returned value of `replace` (not using a `receive`, because those cannot test the returned value of `replace`), similar to the tests at lines 1682-1689.

The tests at line 1682-1689 tested:
- The returned value of a `replace(k, null)` when `k` is in the map and so, `k` gets deleted and `replace` returns `true`. A follow up test should be added `get(k)` that should return `null` to confirm that the `k` is deleted.
- The returned value of a `replace(k, null)` when `k` is not in the map and never was in the map. A follow up test should be added `get(k)` to confirm that it returns `null`.
- The returned value of a `replace(k, v)` (for `v` not null) when `k` is not in the map because it was previously deleted by `replace(k, null)`. This returned `true` (which is puzzling to me, because `k` is not in the map). An alternative test should be added in which a key is deleted using `del(k)` and immediately followed by `get(k)` (which returns `null`), then followed by `replace(k, v)` (which should return `false`), and then followed by `get(k)` (which should return `null`).
- The returned value of a `replace(k, v)` (for `v` not null) when `k` is not in the map and never was in the map. This returned `false`.
- A test that is missing is: `replace(k, v)` (for `v` not null) when `k` is actually in the map. This should return `true`.

In theory, similar tests should be added for the rest of maps, not just for `IntMap1`.

Reply:
yes, makes sense, I'll add tests for all maps